WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReaders/CardReaderSettingsViewModel.swift

```diff
@@ -59,15 +59,25 @@ final class CardReaderSettingsViewModel: ObservableObject {
 
         let siteID = ServiceLocator.stores.sessionManager.defaultStoreID ?? Int64.min
 
-        let action = CardPresentPaymentAction.startCardReaderDiscovery(siteID: siteID, onCompletion: { cardReaders in
+        let discoveryAction = CardPresentPaymentAction.startCardReaderDiscovery(siteID: siteID, onCompletion: { cardReaders in
             // TODO. To be implemented
             // Leaving these prints here to help test the PR
             print("==== new card readers discovered begins ====")
             print("new readers: ", cardReaders)
             print("==== new card readers discovered ends   ====")
         })
 
-        ServiceLocator.stores.dispatch(action)
+        ServiceLocator.stores.dispatch(discoveryAction)
+
+        let discoveryCancellationAction = CardPresentPaymentAction.cancelCardReaderDiscovery { status in
+            // TODO. To be implemented
+            // Leaving these prints here to help test the PR
+            print("===== cancellation begins ====")
+            print("new status ", status)
+            print("===== cancellation ends ====")
+        }
+
+        ServiceLocator.stores.dispatch(discoveryCancellationAction)
```

Comment:
Is the intention here to launch the discovery action and immediately cancel the action? or am I missing something?

Reply:
This is just a completely made up flow to start the discovery process and trigger a cancellation right away.